UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.UnitTests/Controllers/WebhookControllerTest.cs

```diff
@@ -0,0 +1,65 @@
+﻿using FakeItEasy;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using UKHO.ExternalNotificationService.API.Controllers;
+
+namespace UKHO.ExternalNotificationService.API.UnitTests.Controllers
+{
+    [TestFixture]
+    public class WebhookControllerTest
+    {
+        private WebhookController _controller;
+        private ILogger<WebhookController> _fakeLogger;
+        private IHttpContextAccessor _fakeHttpContextAccessor;
+
+        [SetUp]
+        public void Setup()
+        {
+            _fakeLogger = A.Fake<ILogger<WebhookController>>();
+            _fakeHttpContextAccessor = A.Fake<IHttpContextAccessor>();
+
+            _controller = new WebhookController(_fakeHttpContextAccessor, _fakeLogger);
+        }
+
+        [Test]
+        public void WhenValidHeaderRequestedInNewFilesPublishedOptions_ThenReturnsOkResponse()
+        {
+            _controller.ControllerContext.HttpContext = new DefaultHttpContext();
+            _controller.HttpContext.Request.Headers.Add("WebHook-Request-Origin", "test.example.com");
+
+            var result = (StatusCodeResult)_controller.Options();
+
+            Assert.AreEqual(StatusCodes.Status200OK, result.StatusCode);
```

Comment:
Can we also verify that the correct headers have been set for the Options handshake please?  i.e.:
- WebHook-Allowed-Rate == "*"
- WebHook-Allowed-Origin == "test.example.com"

Reply:
Perfect, thanks